src/Services/ConsultationService.php

```diff
@@ -25,4 +29,53 @@ public function getConsultationsList(array $search = []): Builder
             $criteria
         );
     }
+
+    public function store(array $data = []): Consultation
+    {
+        DB::beginTransaction();
+        try {
+            $consultation = $this->consultationRepositoryContract->create($data);
+            DB::commit();
```

Comment:
Does `->create()` in consultation repository do anything more than creating one record in consultation table? Because if not, you don't need explicit DB transactions here...

And also, laravel 8+ let's you write this simply as
```
DB::transaction(function () {
    $consultation = $this->consultationRepositoryContract->create($data);
});
```
without manual ::commit() and ::rollBack()